### PR TITLE
fix: optimise initial rendering time in analytics overview

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
- Issue Type - Bug

- Description - The analytics overview page was making all the requests and rendering all its children components as soon as the component mounts which increases the total load time for the page. Added ids to all the components for conditional rendering for components that are once present in the viewport.

- Please look into the following PR for further reference - appbaseio-confidential/arc-dashboard#327

### CODE REVIEW

File: components/analytics/components/Analytics.js [hunk 1]
```diff
@@ -61,6 +61,47 @@ const Analytics = ({
...
+		'popular-filters-component': false,
+		'popular-results-component': false,
+		'popular-searches-component': false,
+		'request-distribution-component': false,
+		'search-latency-component': false,
+		'search-volume-component': false,
+		'summary-component': true,
+	});
+
+	const inViewPort = (id) => {
```
Comment (bietkul) [inline components/analytics/components/Analytics.js:83]: Put it in util file

File: components/analytics/components/Analytics.js [hunk 2]
```diff
@@ -77,86 +118,121 @@ const Analytics = ({
 				}
 				css="margin-bottom: 20px"
 				title="Summary"
+				id="summary-component"
 			>
-				<Summary filterId={filterId} />
+				{visibility['summary-component'] && <Summary filterId={filterId} />}
```
Comment (bietkul) [inline components/analytics/components/Analytics.js:123]: Use the div with id

File: components/analytics/components/Analytics.js [hunk 3]
```diff
@@ -77,86 +118,121 @@ const Analytics = ({
...
+					<Flex
+						flexDirection="column"
+						css="width: 100%;margin-top: 20px"
+						id="request-distribution-component"
+					>
+						{visibility['request-distribution-component'] && (
+							<RequestDistribution filterId={filterId} displayFilter={false} />
+						)}
 					</Flex>
 				</React.Fragment>
```
Comment (bietkul) [inline components/analytics/components/Analytics.js:237]: Add request logs component at here

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Move the views associated with advanced analytics endpoint to a single component and render it after summary with a unique id.

**Comment by anik-ghosh-au7** [comment]
@bietkul I have done the changes
https://www.loom.com/share/a2d93d7ec3fd4497a0cb5bc11bf3d427

**Comment by anik-ghosh-au7** [comment]
@bietkul 

demo updated version - 

https://www.loom.com/share/34ee743493424544a91c236c17f3f7f6

File: components/analytics/AdvancedAnalytics.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+AdvancedAnalytics.propTypes = {
+	searchVolume: PropTypes.array,
+	popularSearches: PropTypes.array,
+	noResults: PropTypes.array,
+	popularResults: PropTypes.array,
+	popularFilters: PropTypes.array,
+	plan: PropTypes.string.isRequired,
+	displayReplaySearch: PropTypes.bool,
+	handleReplaySearch: PropTypes.func.isRequired,
+	onClickViewAll: PropTypes.shape({
```
Comment (bietkul) [inline components/analytics/AdvancedAnalytics.js:150]: @anik-ghosh-au7 Is that prop added by you? If yes then why?

File: components/analytics/components/Analytics.js [hunk 5]
```diff
@@ -49,18 +23,44 @@ const handleClickBar = (payload) => {
...
 	loading,
 	onClickViewAll,
 	displayReplaySearch,
 	handleReplaySearch,
 	filterId,
+	appName,
 }) => {
+	useEffect(() => {
+		window.addEventListener('scroll', tracker);
+		const initTracker = setTimeout(() => {
```
Comment (bietkul) [inline components/analytics/components/Analytics.js:36]: @anik-ghosh-au7 Why are you using `setTimeout`?

File: components/analytics/components/Analytics.js [hunk 6]
```diff
@@ -49,18 +23,44 @@ const handleClickBar = (payload) => {
...
+			tracker();
+		}, 0);
+		return () => {
+			window.removeEventListener('scroll', tracker);
+			if (initTracker) {
+				clearTimeout(initTracker);
```
Comment (bietkul) [inline components/analytics/components/Analytics.js:42]: @anik-ghosh-au7 Revisit this too.

File: components/analytics/index.js [hunk 7]
```diff
@@ -117,12 +71,6 @@ Main.propTypes = {
 	plan: PropTypes.string.isRequired,
 	displayReplaySearch: PropTypes.bool,
 	fetchAppAnalytics: PropTypes.func.isRequired,
```
Comment (bietkul) [inline components/analytics/index.js:73]: @anik-ghosh-au7 Why is it there? Do you not see lint errors?

File: components/analytics/index.js [hunk 8]
```diff
@@ -131,25 +79,9 @@ Main.propTypes = {
...
 	return {
 		plan: 'growth',
 		appName: get(state, '$getCurrentApp.name'),
-		popularSearches: get(appAnalytics, 'popular_searches', []),
-		popularResults: get(appAnalytics, 'popular_results', []),
-		popularFilters: get(appAnalytics, 'popular_filters', []),
-		searchVolume: get(appAnalytics, 'search_histogram', []),
-		noResults: get(appAnalytics, 'no_results_searches', []),
-		isFetching: get(state, '$getAppAnalytics.isFetching'),
 		filters: get(state, `$getSelectedFilters.${props.filterId}`),
```
Comment (bietkul) [inline components/analytics/index.js:85]: @anik-ghosh-au7 Where are you using `filters` in this component?

**Review by bietkul — CHANGES_REQUESTED** [review]

**Review by bietkul — APPROVED** [review]

**Pull request merged**